RevenueCatUI/Templates/Components/Button/ButtonComponentView.swift

```diff
@@ -28,13 +31,74 @@ struct ButtonComponentView: View {
 
     var body: some View {
         Button(
-            action: { viewModel.onClick() },
+            action: { onClick() },
             label: { StackComponentView(viewModel: viewModel.stackViewModel) }
-        )
+        ).sheet(isPresented: .isNotNil($inAppBrowserURL)) {
+            SafariView(url: inAppBrowserURL!)
+        }
+    }
+
+    private func onClick() {
+        switch viewModel.component.action {
+        case .restorePurchases:
+            // swiftlint:disable:next todo
+            // TODO handle restoring purchases
+            break
+        case .navigateTo(let destination):
+            navigateTo(destination: destination)
+        case .navigateBack:
+            // swiftlint:disable:next todo
+            // TODO handle navigating back
+            break
+        }
+    }
+
+    private func navigateTo(destination: PaywallComponent.ButtonComponent.Destination) {
+        switch destination {
+        case .customerCenter:
+            // swiftlint:disable:next todo
+            // TODO handle navigating to customer center
+            break
+        case .URL(let urlLid, let method),
+                .privacyPolicy(let urlLid, let method),
+                .terms(let urlLid, let method):
+            navigateToUrl(urlLid: urlLid, method: method)
+        }
+    }
+
+    private func navigateToUrl(urlLid: String, method: PaywallComponent.ButtonComponent.URLMethod) {
+        guard let urlString = try? viewModel.localizedStrings.string(key: urlLid),
+        let url = URL(string: urlString) else {
+            Logger.error(Strings.paywall_invalid_url(urlLid))
+            return
+        }
```

Comment:
2 comments for your consideration: 
- This means that any error in configuring the URLs will only surface once the button is clicked. Alternatively, we can parse the URL at construction time.
- This bit of logic in the view means it's no longer a "dumb view", as MVVM prescribes. Alternatively, it could be moved to the ViewModel.

Let me know what you think!

Reply:
Instead of calling this in `navigateToUrl`, can we call this in the constructor of the `ButtonComponentViewModel`?

The constructor of the view models "throw" errors. So you could do something like how the `PurchaseButtonComponentViewModel` does 👇 

```swift
self.cta = try localizedStrings.string(key: component.cta)
self.ctaIntroOffer = try component.ctaIntroOffer.flatMap {
    try localizedStrings.string(key: $0)
}
```

You can try to grab and parse the URL (if it exists for the action) in the constructor and then throw an error to 💥 the paywall on initialization.

This _may_ require another type of view model enum or something but I think the best way to follow the standard error flow of paywalls to error as early as possible 😇 

Let me know if you have any questions on this! I'd be happy to pair 😊

Reply:
Yea this makes total sense! Thanks for the pointers too! I made the change here, and added some GitHub comments explaining my rationale: https://github.com/RevenueCat/purchases-ios/pull/4360/commits/63635a6190d858364200bffd28b2482e61fe7b0f.